### PR TITLE
Retry for network changes on localhost

### PR DESCRIPTION
**Pull request by edlerd**
## Done

- requests to alter networks on localhost will fail, add retry logic for those cases. The problem only occurs when talking directly to lxd (without dev server) on localhost via https://localhost:8443/ui/

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - check network editing/renaming/creation/deletion
    - the error can only be triggered when talking directly to LXD on localhost (without the dev server), maybe not worth to find a way to set this up, we can verify the fix from the nightly LXD build after it is merged.

### CODE REVIEW

**Review by lorumic — COMMENTED** [review]
The fix makes sense, and I expect it to work even if I cannot test it, but it's a somewhat ugly workaround. Do we have an idea of what is the root cause of this problem? How was this found? Might be worth discussing it with the core team?

File: src/api/networks.tsx [hunk 1]
```diff
@@ -119,17 +140,39 @@ export const renameNetwork = (
...
+            resolve(renamedNetwork);
+          }
+        }
+        reject(e);
+      });
   });
 };
 
-export const deleteNetwork = (name: string, project: string) => {
+export const deleteNetwork = (name: string, project: string): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (edlerd) [reply src/api/networks.tsx:157]: In this case yes, because we are calling `resolve();` below.

**Review by lorumic — APPROVED** [review]
I will approve, but I still suggest to consider what I wrote in the previous comment:

> The fix makes sense, and I expect it to work even if I cannot test it, but it's a somewhat ugly workaround. Do we have an idea of what is the root cause of this problem? How was this found? Might be worth discussing it with the core team?

**Comment by edlerd** [comment]
> I will approve, but I still suggest to consider what I wrote in the previous comment:
> 
> > The fix makes sense, and I expect it to work even if I cannot test it, but it's a somewhat ugly workaround. Do we have an idea of what is the root cause of this problem? How was this found? Might be worth discussing it with the core team?

Opened an issue https://github.com/canonical/lxd/issues/12475

**Comment by lorumic** [comment]
> Opened an issue [canonical/lxd#12475](https://github.com/canonical/lxd/issues/12475)

Well done, thanks!

**Comment by edlerd** [comment]
Merged as upstream has a hard time finding a fix and we want the UI to be without this bug.

**Pull request merged**